deploy.md

```diff
@@ -0,0 +1,677 @@
+Typically, non-prod environments like dev will use something like `dev.stingray.example.com`
+and prod will use something like `stingray.example.com`
+
+## Git Repository Setup
+
+Create a Git repository for your FilmDrop Infrastructure deployment
+project (e.g., `filmdrop-{project_name}-aws-infra-ops`). This will be the
+FilmDrop deployment project that will deploy FilmDrop from CI.
+
+Copy the contents of the `project` directory from this
+repository to the root of your new repository.
+
+Rename `remove.gitignore` to `.gitignore`
+
+Commit this code to the new repository as the branch `main`. This will start
+a pipeline, which will probably fail since we haven't yet performed some required
+configuration.
+
+Before making any other changes, create a new branch to put these changes in.
+
+## Boostrap AWS Resources
+
+### Create Deployment IAM Role
+
+The deployment IAM Role is used when deploying FilmDrop infrastructure from CI.
+The deployment Role should be created once per AWS Account.
+
+There are two different CloudFormation templates, depending on which CI system
+you are using:
+
+- GitLab: bootstrap/gitlab_deploy_role_cfn.yml
+- GitHub: bootstrap/github_deploy_role_cfn.yml
+
+You can create a CloudFormation Stack from these templates with the
+following command (GitLab example shown), The Terraform state S3 Bucket name
+(discussed later) must be globally unique, so it is recommended to use where the value
+passed for `TerraformStateBucketName` is a unique bucket name based on the template
+`filmdrop-{project_name}-{region}-terraform-state-{random_string}`:
+
+```shell
+aws cloudformation deploy --stack-name "appFilmDropDeployRoleBootstrap" \     
+   --template-file bootstrap/gitlab_deploy_role_cfn.yml \
+   --capabilities=CAPABILITY_NAMED_IAM \
+   --parameter-overrides     TerraformStateBucketName=filmdrop-stingray-us-west-2-terraform-state-20240515
+```
+
+Copy the Role ARNs created by this deploy and either:
+
+- GitLab: Enable the trust policy for these roles in GitLab Runner. After they
+are enabled, GitLab CI pipelines of your project will be able to use
+this role to connect to corresponding AWS accounts.
+- GitHub: TBD
+
+### Create OpenSearch Service Service Role
+
+The Service Role should be created only once per AWS Account.
+
+OpenSearch requires an IAM Service Role named `AWSServiceRoleForAmazonOpenSearchService` to be defined.
+
+This role is automatically created when creating an OpenSearch Service domain
+using the AWS Console, but must be created explicitly when creating a domain
+programmatically.
+
+Note that this will fail if the Role already exists, which is fine, since deployment
+of the stac-server module only requires that it exists.
+
+To deploy, run:
+
+```shell
+cd bootstrap
+terraform init
+terraform apply -target=aws_iam_service_linked_role.opensearch_linked_role
+```
+
+There is no need to keep the terraform state file, as running this is intended to
+be an idempotent operation.
```

Comment:
You could just `aws iam create-service-linked-role --aws-service-name opensearchservice.amazonaws.com`?

Reply:
I found I also had to `aws iam create-service-linked-role --aws-service-name wafv2.amazonaws.com` to make something using the WAF happy.